experiments/AMIP/modular/components/ocean/slab_ocean_init.jl

```diff
@@ -128,3 +129,12 @@ end
 step!(sim::SlabOceanSimulation, t) = step!(sim.integrator, t - sim.integrator.t, true)
 
 reinit!(sim::SlabOceanSimulation) = reinit!(sim.integrator)
+
+"""
+    get_model_state_vector(sim::SlabOceanSimulation)
+
+Extension of Checkpointer.get_model_state_vector to get the model state.
+"""
+function get_model_state_vector(sim::SlabOceanSimulation)
+    return sim.integrator.u
```

Comment:
I'm not exactly familiar with these containers - above, we `return sim.integrator.u.bucket` with the `BucketSimulation` - is there a similar `sim.integrator.u.slabocean` or is this indeed correct? In which case is it useful to have a fallback method dispatching on the abstract type (?)

Reply:
It's correct. Each component model has its own `_init` file where it specifies model-specific paths to states and cache. I've tested it locally and it ran, but can add it to a buildkite driver in the long runs as well though.